convert/convert.go

```diff
@@ -49,80 +49,105 @@ func File(file *hcl.File, options Options) ([]byte, error) {
 
 type jsonObj map[string]interface{}
 
-func convertFile(file *hcl.File, options Options) (jsonObj, error) {
-	c := converter{bytes: file.Bytes, options: options}
-	body := file.Body.(*hclsyntax.Body)
-
-	return c.convertBody(body)
-}
-
 type converter struct {
 	bytes   []byte
 	options Options
 }
 
-func (c *converter) rangeSource(r hcl.Range) string {
-	// for some reason the range doesn't include the ending paren, so
-	// check if the next character is an ending paren, and include it if it is.
-	end := r.End.Byte
-	if c.bytes[end] == ')' {
-		end++
+func convertFile(file *hcl.File, options Options) (jsonObj, error) {
+	body, ok := file.Body.(*hclsyntax.Body)
+	if !ok {
+		return nil, fmt.Errorf("convert file body to body type")
 	}
-	return string(c.bytes[r.Start.Byte:end])
+
+	c := converter{
+		bytes:   file.Bytes,
+		options: options,
+	}
+
+	out, err := c.convertBody(body)
+	if err != nil {
+		return nil, fmt.Errorf("convert body: %w", err)
+	}
+
+	return out, nil
 }
 
 func (c *converter) convertBody(body *hclsyntax.Body) (jsonObj, error) {
-	var err error
 	out := make(jsonObj)
-	for key, value := range body.Attributes {
-		out[key], err = c.convertExpression(value.Expr)
-		if err != nil {
-			return nil, err
+
+	for _, block := range body.Blocks {
```

Comment:
Does the order of converting bodies and attributes matter?

Reply:
It does not, no. This must have just gotten moved as I was trying to make sense of the codebase and how the HCL parser bucketed each part.